common/gce.py

```diff
@@ -88,3 +88,56 @@ def get_base_target_link(experiment_config):
 def get_instance_from_preempted_operation(operation, base_target_link) -> str:
     """Returns the instance name from a preempted |operation|."""
     return operation['targetLink'][len(base_target_link):]
+
+
+def get_instance_group_managers():
+    """Returns the instance group managers resource."""
+    return thread_local.service.instanceGroupManagers()
+
+
+def get_instance_group_size(instance_group: str, project: str,
+                            zone: str) -> int:
+    """Returns the number of instances running in |instance_group|."""
+    managers = get_instance_group_managers()
+    request = managers.get(instanceGroupManager=instance_group,
+                           project=project,
+                           zone=zone)
+    return request.execute()['targetSize']
+
+
+def resize_instance_group(size, instance_group, project, zone):
+    """Changes the number of instances running in |instance_group| to |size|."""
+    assert size >= 1
+    managers = get_instance_group_managers()
+    request = managers.resize(instanceGroupManager=instance_group,
+                              size=size,
+                              project=project,
+                              zone=zone)
+    return request.execute()
+
+
+def delete_instance_group(instance_group, project, zone):
+    """Deletes |instance_group|."""
+    managers = get_instance_group_managers()
+    request = managers.delete(instanceGroupManager=instance_group,
+                              zone=zone,
+                              project=project)
+    return request.execute()
+
+
+def create_instance_group(name: str, instance_template_url: str,
+                          experiment: str, project: str, zone: str):
+    """Creates an instance group named |name| from the template specified by
+    |instance_template_url|."""
+    managers = get_instance_group_managers()
+    target_size = 1
+    base_instance_name = 'w-' + experiment
```

Comment:
what does w stand for ?
probably add '-' after experiment

Reply:
w stands for worker. I'm going to change it to "m" for measurer and leave a comment about this.
GCE adds the hyphen at the end. No need.